src/main/webapp/script.js

```diff
@@ -122,15 +122,15 @@ function createIndicator(dataSlideNumber, className) {
 function getNextPotentialMatch() {
   deletePotentialMatchInfo();
   let currentUser = getCurrentUserId();
-  fetch('/potential-matches?userid=' + currentUser).then(response => response.text()).then((pmID) => { 
+  fetch('/potential-matches?userid=' + currentUser).then(response => response.json()).then((pmID) => { 
       let noMatch = "NO_POTENTIAL_MATCHES";
```

Comment:
nit: pull this into a `const` defined at/near the top of the file and add a comment that it must be kept in-sync with XServlet.java

Reply:
(if the servlet isn't merged yet, feel free to address later when it is)